rfcs/0007-filtering.md

```diff
@@ -0,0 +1,93 @@
+(10 out of 69 top-level getters; 6 out of 42 fields in the ItemComponent), it makes
+sense for the server to send us only the data consumed.  This reduces bandwidth requirements for receiving the data.
+
+For example, GraphQL solves this problem by allowing clients to specify field selection (e.g. a database projection) at
+invocation time.
+
+While not as powerful as the GraphQl offering where one can field select as deep in the hierarchy as desired, FHIR does offer a top level projection through the use of the _elements parameter supported both in R3 and R4.
+
+## Proposal
+[Proposal]: #proposal
+
+FHIR section 3.1.1.5.9 specifies the behavior of _elements as follows:
+```Servers are not obliged to return just the requested elements. Servers SHOULD always return mandatory elements whether they are requested or not. Servers SHOULD mark the resources with the tag SUBSETTED to ensure that the incomplete resource is not actually used to overwrite a complete resource.```
+
+This proposal changes `not obliged` to `obliged` in the FHIR specification for the **fhir/ExplanationOfBenefit** endpoint, thus providing a guarantee of the performance saving behavior.  The mandatory elements and tag marking behavior are unchanged.  This proposal does not address reducing the number of fields returned in ItemComponent as the FHIR specification makes no provision for selecting fields of contained objects.
+
+### Proposed Solution: Unresolved Questions
+[Proposed Solution: Unresolved Questions]: #proposed-solution-unresolved-questions
+
+How much will this save us?
+
+As part of a larger effort to reduce job times from days to minutes, AB2D is constructing a performance test harness
+which will at its basic element, perform a single EOB retrieval.  Anecdotally, BFD EOB retrievals have been calibrated
+in the 500 msec range.  This harness will allow a more deliberate measurement of the actual response times, independent
+of the current AB2D architecture.  Once the basic harness is ready, before and after retrieval times can be obtained to
+provide a firm answer to the question.
+
+Additional goals for the harness are to assess how much can concurrency increase.  Currently, AB2D limits EOB
+retrievals to 32 concurrent requests.  The plans are to run cooperative experiments to establish a maximum concurrent
+limit.
+
+The harness will make it much more convenient to characterize current performance and to confirm or disprove any
+expected benefits of changes.
+
+Proposed profiling is invoking BFD with at least 10,000 EOB Requests by Patient ID (ExplanationOfBenefit?patient=) and to measure
+the aggregate time of completion of all calls.  In other words, sum up the time taken by each call over all calls.
+
+### Proposed Solution: Drawbacks
+[Proposed Solution: Drawbacks]: #proposed-solution-drawbacks
+
+This solution incurs a nominal amount of extra processing to traverse the EOB object and null out fields that do not
+need to be returned.
```

Comment:
I think this could be done without having to take this approach, but may take some re-engineering of the current EOB processing.